packages/google_adsense/lib/src/adsense_web.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:web/web.dart' as web;
+import 'ad_unit_widget.dart';
+import 'ad_unit_widget_web.dart';
+
+/// Main class to work with the library
+class Adsense {
+  /// Returns a singleton instance of Adsense library public interface
+  factory Adsense() => _instance ?? Adsense._internal();
+
+  Adsense._internal() {
+    _instance = this;
+  }
+
+  static Adsense? _instance = Adsense._internal();
+  bool _isInitialized = false;
+  String _adClient = '';
+  static const String _url =
+      'https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-';
+
+  /// Initialization API. Should be called ASAP, ideally in the main method of your app.
+  /// Throws [StateError] if called more than once
+  void initialize(String adClient) {
+    if (_isInitialized) {
+      throw StateError('Adsense was already initialized, skipping');
+    }
+    _adClient = adClient;
+    _addMasterScript(_adClient);
+    _isInitialized = true;
+  }
+
+  /// Returns a configurable [AdUnitWidget]
+  AdUnitWidget adUnit(
+      {required String adSlot,
+      String adClient = '',
+      bool isAdTest = false,
+      Map<String, dynamic> adUnitParams = const <String, dynamic>{},
+      String? cssText}) {
+    return AdUnitWidgetWeb(
+        adSlot: adSlot,
+        adClient: adClient.isNotEmpty ? adClient : _adClient,
+        isAdTest: isAdTest,
+        additionalParams: adUnitParams,
+        cssText: cssText);
+  }
+
+  void _addMasterScript(String adClient) {
+    final web.HTMLScriptElement adsbygoogle = web.HTMLScriptElement();
+    adsbygoogle.innerText = 'adsbygoogle = window.adsbygoogle || [];';
+    final web.HTMLScriptElement script =
+        web.document.createElement('script') as web.HTMLScriptElement
+          ..async = true
+          ..crossOrigin = 'anonymous';
+    script.src = _url + adClient;
+    (web.document.head ?? web.document).appendChild(adsbygoogle);
+    (web.document.head ?? web.document).appendChild(script);
```

Comment:
`web.document` is a bad default: calling `appendChild` on it is going to fail (try it in the JS console: `document.appendChild(document.createElement('div'))`.

In this case, I'd do:

```dart
web.document.head!.appendChild...
```

Instead of defaulting to `document`.